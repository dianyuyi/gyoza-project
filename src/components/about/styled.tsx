import tw, { styled } from 'twin.macro'
import { up } from 'styled-breakpoints'

export const Block = styled.section`
  ${tw`mt-0 pt-0 relative pb-8`}

  ${up('md')} {
    margin-top: -16vw;
    padding-top: 12vw;
  }
`

export const ContentWrapper = styled.div`
  ${tw`bg-white flex flex-col items-center`}
`

export const ImageContent = styled.div`
  ${tw`w-full md:w-1/2`}
`

export const TextContent = styled.div`
  ${tw`w-auto box-border px-[10vw] pt-[4vh] pb-[12vh]`}
`

export const BlockTitle = styled.h2`
  ${tw`pl-12 text-lg tracking-wide absolute`}

  &:before {
    content: '';
    ${tw`absolute block w-12 h-px bg-[#C5CBD1] top-1/2 left-0`}
  }
`

export const MainContext = styled.div`
  white-space: pre-line;
  ${tw`text-base leading-8 mt-6`};
`

export const SubContext = styled.div`
  ${tw`mt-4 text-xs tracking-wider	leading-6`}
`
