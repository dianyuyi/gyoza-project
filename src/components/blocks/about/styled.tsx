import tw, { styled } from 'twin.macro'
import { up } from 'styled-breakpoints'
import { HorizontalTitle } from 'src/components/title'

export const Block = styled.section`
  ${tw`mt-0 pt-0 relative mt-12 pb-16`}

  ${up('md')} {
    margin-top: -16vw;
    padding-top: 12vw;
  }
`

export const ContentWrapper = styled.div`
  ${tw`bg-white flex flex-col items-center md:flex-row`}
`

export const ImageContent = styled.div`
  ${tw`w-full md:w-1/2`}
`

export const TextContent = styled.div`
  ${tw`w-auto box-border px-[10vw] py-[16vh]`}
`

export const BlockTitle = styled(HorizontalTitle)`
  ${tw`pl-12`}
`

export const MainContext = styled.div`
  white-space: pre-line;
  ${tw`text-base leading-8 mt-6`};
`

export const SubContext = styled.div`
  ${tw`mt-4 text-xs tracking-wider	leading-6`}
`
