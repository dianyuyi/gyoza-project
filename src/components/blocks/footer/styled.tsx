import tw, { styled } from 'twin.macro'

export const StyledFooter = styled.footer`
  ${tw`bg-[#222] text-white mt-8`}
`

export const Container = styled.div`
  ${tw`container grid grid-cols-2 gap-4 place-content-center py-8 mx-auto`}
`

export const InfoBox = styled.div`
  ${tw`flex flex-col justify-center items-center`}
`

export const Title = styled.p`
  ${tw`text-xl`}
`
export const Text = styled.p`
  ${tw`text-left`}
`
