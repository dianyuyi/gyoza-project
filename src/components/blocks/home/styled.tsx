import tw, { styled } from 'twin.macro'

type StyleProps = {
  desktop: boolean | null | undefined
}

export const StoreContainer = styled.section`
  ${tw`relative w-screen h-[110vh] overflow-hidden sm:h-[130vh] lg:h-[120vh] bg-[#f2f0ed]`}
`

export const StoreName = styled.div`
  ${tw`absolute w-32 top-6 left-6 z-50 md:left-48 md:top-24`}
`

export const ChineseTitle = styled.div`
  ${tw`w-full leading-10 text-2xl tracking-widest	`}

  p {
    ${tw`mb-7`}
  }
`

export const EngTitle = styled.div`
  ${tw`w-full leading-10 text-xs tracking-widest`}
`

export const LeftWrapper = styled.div(({ desktop }: StyleProps) => [
  tw`absolute overflow-hidden -right-1/3 top-8 md:right-0`,
  desktop && tw`bottom-8 hidden md:block right-auto top-auto`,
])

export const CenterWrapper = styled.div(({ desktop }: StyleProps) => [
  tw`absolute overflow-hidden -translate-x-1/2 z-10 left-[60%] top-[24vh] lg:left-1/2 lg:top-[16vh]`,
  desktop && tw`left-[35%] top-[40vh] z-0 hidden md:block lg:left-1/3 lg:top-auto lg:bottom-[12vh]`,
])

export const RightWrapper = styled.div(({ desktop }: StyleProps) => [
  tw`absolute overflow-hidden -left-1/4 bottom-4 md:-left-8 md:z-20 md:-bottom-8 lg:bottom-2`,
  desktop && tw`top-48 hidden md:block md:-left-24`,
])

export const BottomIntro = styled.h2`
  ${tw`absolute right-8 bottom-0 text-sm vertical-rl z-[100] md:right-[40%] md:bottom-24 lg:right-1/4`}
`

export const IntroSentence = styled.p`
  ${tw`mx-2`}
`
export const IntroChar = styled.span`
  ${tw`my-1 font-light`}
`
