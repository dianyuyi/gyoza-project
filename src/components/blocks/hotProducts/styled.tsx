import tw, { styled } from 'twin.macro'
import { up } from 'styled-breakpoints'

import { VerticalTitle, HorizontalTitle } from 'src/components/title'
import { ViewMoreLink } from 'src/components/link'

export const Block = styled.section`
  ${tw`mt-0 pt-0 relative mt-12 pb-16 `}
`
export const TitleWrapper = styled.div`
  ${tw`w-12 block md:w-full lg:max-w-5xl lg:mx-auto`}
`

export const Title = styled(VerticalTitle)`
  ${tw`ml-[10vw] md:absolute`}
`
export const Subtitle = styled(HorizontalTitle)`
  ${tw`top-0 right-[16vw] pl-12 md:left-[20vw] md:right-auto lg:left-[32vw]`}
`

export const MoreLink = styled(ViewMoreLink)`
  ${tw`-top-1 right-0 hidden md:inline-block md:right-12 lg:right-[16vw]`}
`

export const ContentWrapper = styled.div`
  ${tw`flex flex-col items-center md:flex-row md:pl-[10vw] md:pt-[16vh] lg:max-w-5xl lg:mx-auto lg:pl-[16vw]`}
`

export const ImageContent = styled.div`
  ${tw`w-full md:w-1/2`}
`

export const TextContent = styled.div`
  ${tw`w-auto box-border px-[10vw] pt-[4vh] pb-[12vh]`}
`
