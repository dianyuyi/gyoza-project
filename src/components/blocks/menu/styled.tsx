import tw, { styled } from 'twin.macro'

export const PrintButton = styled.button`
  ${tw`fixed p-2 bg-[#222] text-white text-sm w-12 top-1/2 rounded-sm`}
`

export const Block = styled.section`
  ${tw`m-auto p-4 container
    print:w-[21cm] print:h-[29.7cm] print:p-0 print:mt-4
  `}
`

export const GridContainer = styled.div`
  ${tw`bg-white grid grid-cols-3 gap-x-4 gap-y-12`}
`

export const MenuTitle = styled.h2`
  ${tw`text-5xl mx-auto py-8 text-center`}
`

export const GroupTitle = styled.div`
  ${tw`bg-[#222] text-white text-center p-4 py-2 mb-4 text-lg tracking-widest font-bold`}
`

export const GroupBox = styled.div`
  ${tw`block`}
`
