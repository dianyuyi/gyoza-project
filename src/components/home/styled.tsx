import tw, { styled } from 'twin.macro'

export const StoreContainer = styled.section`
  ${tw`relative w-screen h-screen overflow-hidden`}
`

export const StoreName = styled.div`
  ${tw`absolute w-32 top-6 left-6 z-10`}
`

export const ChineseTitle = styled.div`
  ${tw`w-full leading-10 text-2xl tracking-widest	`}

  p {
    ${tw`mb-8`}
  }
`

export const EngTitle = styled.div`
  ${tw`w-full leading-10 text-xs tracking-widest	`}
`

export const LeftWrapper = styled.div`
  ${tw`absolute overflow-hidden -right-1/3 top-6`}
`

export const CenterWrapper = styled.div`
  ${tw`absolute overflow-hidden left-1/2 -translate-x-1/2 top-1/4`}
`

export const RightWrapper = styled.div`
  ${tw`absolute overflow-hidden -left-1/4 bottom-1/4`}
`
