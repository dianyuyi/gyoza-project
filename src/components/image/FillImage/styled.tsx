import tw, { styled, css } from 'twin.macro'

interface Props {
  width: number | string
  height: number | string
}
export const Box = styled.div(({ width, height }: Props) => [
  tw`relative w-auto h-auto`,
  width &&
    css`
      width: ${width};
    `,
  height &&
    css`
      height: ${height};
    `,
])
