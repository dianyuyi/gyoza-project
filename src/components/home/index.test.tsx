/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import '@testing-library/jest-dom'
// import { StoreContainer } from './styled'

import styled from 'styled-components'
// import tw from 'twin.macro'

const StoreContainer = styled.div`
  display: block;
`

test('it works', () => {
  const { container } = render(<StoreContainer />)
  expect(container).toMatchSnapshot()
})
