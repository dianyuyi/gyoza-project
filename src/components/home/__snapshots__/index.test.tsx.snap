// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  display: block;
}

<div>
  <div
    class="c0"
  />
</div>
`;
